PyInstaller/building/utils.py

```diff
@@ -95,11 +95,24 @@ def add_suffix_to_extensions(toc):
     new_toc = TOC()
     for inm, fnm, typ in toc:
         if typ == 'EXTENSION':
+            if is_py3:
+                # Change the dotted name into a relative path. This places C
+                # extensions in this Python-standard location. This only works
+                # in Python 3; see comments above
+                # ``sys.meta_path.append(CExtensionImporter())`` in
+                # ``pyimod03_importers``.
+                inm = inm.replace('.', os.sep)
             # In some rare cases extension might already contain a suffix.
             # Skip it in this case.
             if os.path.splitext(inm)[1] not in EXTENSION_SUFFIXES:
-                # Use this file's existing extension.
-                inm = inm + os.path.splitext(fnm)[1]
+                # Determine the base name of the file.
+                if is_py3:
+                    base_name = os.path.basename(inm)
+                else:
+                    base_name = inm.rpartition('.')[2]
+                assert '.' not in base_name
+                # Use this file's existing extension. For extensions such as ``libzmq.cp36-win_amd64.pyd``, we can't use ``os.path.splitext``, which would give only the ```.pyd`` part of the extension.
```

Comment:
Interesting point!

Please wrap the long line.

Reply:
Found only by experimentation -- this surprised me too. Wrapped.